env0/data_cloud_credentials.go

```diff
@@ -0,0 +1,74 @@
+package env0
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var allowedCredentialTypes = []string{
+	"AWS_ASSUMED_ROLE",
+	"AWS_ASSUMED_ROLE_FOR_DEPLOYMENT",
+	"AWS_ACCESS_KEYS_FOR_DEPLOYMENT",
+	"GCP_CREDENTIALS",
+	"GCP_SERVICE_ACCOUNT_FOR_DEPLOYMENT",
+	"AZURE_CREDENTIALS",
+	"AZURE_SERVICE_PRINCIPAL_FOR_DEPLOYMENT",
+}
+
+func dataCloudCredentials() *schema.Resource {
+	allowedCredentialTypesStr := fmt.Sprintf("(allowed values: %s)", strings.Join(allowedCredentialTypes, ", "))
+
+	return &schema.Resource{
+		ReadContext: dataCloudCredentialsRead,
+
+		Schema: map[string]*schema.Schema{
+			"names": {
+				Type:        schema.TypeList,
+				Description: "list of all cloud credentials (by name), optionaly filtered by credential_type",
+				Computed:    true,
+				Elem: &schema.Schema{
+					Type:        schema.TypeString,
+					Description: "the credential name",
+				},
+			},
+			"credential_type": {
+				Type:             schema.TypeString,
+				Description:      "the type of cloud credential to filter by " + allowedCredentialTypesStr,
+				Optional:         true,
```

Comment:
setting to optional incases where someone (for some reason) wants to get all the credentials regardless of type.